design/media_types.go

```diff
@@ -30,6 +30,7 @@ var workItem = a.MediaType("application/vnd.workitem+json", func() {
 	a.Description("A work item hold field values according to a given field type")
 	a.Attribute("id", d.String, "unique id per installation")
 	a.Attribute("version", d.Integer, "Version for optimistic concurrency control")
+	a.Attribute("ExecutionOrder", d.Number, "Order of the workitem in the list")
```

Comment:
?!? What are you doing?

app.WorkItem.Fields[system.Order]
workitem.WorkItem.ExecutionOrder

The problem was, on workitem.WorkItemRepository.Create() you managed to store app.WorkItem.Fields[system.Order] as part of workitem.WorkItem.Fields[system.Order]. It should not be stored in workitem.WorkItem.Fields[system.Order] because it's already stored in workitem.WorkItem.ExecutionOrder, which you did manage to remove from workitem.WorkItem.Fields[system.Order] when you did a workitem.WorkItemRepository.Save()